src/main/java/org/homeschoolpebt/app/interceptor/FlowSwitchInterceptor.java

```diff
@@ -0,0 +1,55 @@
+package org.homeschoolpebt.app.interceptor;
+
+import formflow.library.data.SubmissionRepositoryService;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.util.Objects;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.util.AntPathMatcher;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+@Component
+@Slf4j
+public class FlowSwitchInterceptor implements HandlerInterceptor {
+  public static final String PATH_FORMAT = "/flow/{flow}/{screen}";
```

Comment:
An unfortunate part is that this PATH_FORMAT won't match subflow routes, I don't think. So we'll have to figure out a better approach to this some time later.